pkg/pool-weighted/contracts/test/ManagedPoolStorageLib.t.sol

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import { DSTest } from "ds-test/test.sol";
+import "./Hevm.sol";
+
+import "../managed/ManagedPoolStorageLib.sol";
+
+contract ManagedPoolStorageLibTest is DSTest {
+    Hevm internal constant hevm = Hevm(HEVM_ADDRESS);
+
+    uint256 private constant _WEIGHT_START_TIME_OFFSET = 0;
+    uint256 private constant _WEIGHT_END_TIME_OFFSET = _WEIGHT_START_TIME_OFFSET + _TIMESTAMP_WIDTH;
+    uint256 private constant _SWAP_FEE_START_TIME_OFFSET = _WEIGHT_END_TIME_OFFSET + _TIMESTAMP_WIDTH;
+    uint256 private constant _SWAP_FEE_END_TIME_OFFSET = _SWAP_FEE_START_TIME_OFFSET + _TIMESTAMP_WIDTH;
+    uint256 private constant _SWAP_FEE_START_PCT_OFFSET = _SWAP_FEE_END_TIME_OFFSET + _TIMESTAMP_WIDTH;
+    uint256 private constant _SWAP_FEE_END_PCT_OFFSET = _SWAP_FEE_START_PCT_OFFSET + _SWAP_FEE_PCT_WIDTH;
+    uint256 private constant _SWAP_ENABLED_OFFSET = _SWAP_FEE_END_PCT_OFFSET + _SWAP_FEE_PCT_WIDTH;
+    uint256 private constant _MUST_ALLOWLIST_LPS_OFFSET = _SWAP_ENABLED_OFFSET + 1;
+    uint256 private constant _RECOVERY_MODE_OFFSET = _MUST_ALLOWLIST_LPS_OFFSET + 1;
+
+    uint256 private constant _TIMESTAMP_WIDTH = 32;
+    uint256 private constant _SWAP_FEE_PCT_WIDTH = 62;
+
+    uint256 private constant _MAX_SWAP_FEE = 1 << _SWAP_FEE_PCT_WIDTH;
+
+    function clearWordAtPosition(
```

Comment:
`clearWordAtPosition` and `assertOtherStateUnchanged` would be generally useful. Maybe put them up with WordCodec (solidity-utils/helpers/FoundryHelpers)?

Reply:
Yeah, we'll need them for `ManagedPoolTokenLib`, etc. I'll move these once we add the second test contract which uses them.